content/articles/implementing-a-custom-zoom-in-and-zoom-out-imageview-in-android-with-kotlin/index.md

```diff
@@ -0,0 +1,220 @@
+
+### Importance of including zoom capabilities to android Application.
+- It gives a view of images that are not concentrated onto the current screen.
+- Zoom control enables users to put in place the view to all the large and small objects on the android screen.
+- Zooming give a clear view of the images and objects on android screens.
+- Zoom capabilities provide a way in which users can concentrate on a specified sector of the entire image and study it.
+
+### Prerequisites
+- Ensure you have Android Studio installed
+- Know the fundamental of the kotlin programming language
+- Have basic mathematical knowledge of Matrices concept
+- Have basic knowledge of creating and running android application
+
+### Creating a Customized Zooming 
+To create a customized zooming action, several methods should be implemented for this action to be successful. To begin on this, let us create a class that should extend `AppCompatImageView` and implement the following:
+
+- `View.OnTouchListener`
+- `GestureDetector.OnGestureListener`
+- `GestureDetector.OnDoubleTapListener`
+
+Within the created class ensure you have declared the following variables that would aid in the creation of these zooming controls.
+
+```kotlin
+    var myMatrix: Matrix? = null
+    private var matrixValue: FloatArray? = null
+    var mode = NONE
+
+    // Scales
+    var presentScale = 1f
+    var minimumScale = 1f
+    var maximumScale = 4f
+
+    //Dimensions
+    var originalWidth = 0f
+    var originalHeight = 0f
+    var viewedWidth = 0
+    var viewedHeight = 0
+```
+
+Create the following method which serves as the foundation for the zooming functionality's creation.
+```kotlin
+    private fun constructionDetails(context: Context){
+        super.setClickable(true)
+        myContext=context
+        myScaleDetector= ScaleGestureDetector(context,ScalingListener())
+        myMatrix=Matrix()
+        matrixValue=FloatArray(10)
+        imageMatrix = myMatrix
+        scaleType = ScaleType.MATRIX
+        myGestureDetector = GestureDetector(context, this)
+        setOnTouchListener(this)
+    }
+```
+> Make sure you have declared `myScaleDetector`, `myGestureDetector` and `myContext` variables outside the `onCreate` method.
+
+To retrieve the `ScaleGestureDetector`, which is usually an interface, we'll create an inner class and make sure it implements, `ScaleGestureDetector.SimpleOnScaleGestureListener` This class will then serve as the scale gesture detector.
+
+```kotlin
+private inner class ScalingListener : ScaleGestureDetector.SimpleOnScaleGestureListener(){
+    ...
+}
+```
+
+This class overrides two methods that are `onScaleBegin` and `onScale` which contains the general implementation of the `ScaleGestureDetector` interface.
+
+`onScaleBegin` shows the mode level of the zooming which are ought to be done.
+```kotlin
+        override fun onScaleBegin(detector: ScaleGestureDetector?): Boolean {
```

Comment:
The same to this and the rest where applicable.

Reply:
Hello @Ericgacoki  have made made the changes